unit-tests-java8/src/test/java/org/eclipse/collections/test/RichIterableTestCase.java

```diff
@@ -381,6 +400,137 @@ default void RichIterable_containsAllIterable()
         assertTrue(iterable0.containsAllIterable(Lists.immutable.empty()));
     }
 
+    @Test
+    default void RichIterable_flatCollectBoolean()
+    {
+        RichIterable<BooleanBooleanPair> iterable = this.newWith(
+                PrimitiveTuples.pair(true, false),
+                PrimitiveTuples.pair(false, true));
+
+        BooleanArrayList target = new BooleanArrayList();
+        iterable.flatCollectBoolean((Function<BooleanBooleanPair, BooleanIterable>) each -> BooleanArrayList.newListWith(
+                each.getOne(),
+                each.getTwo()), target);
+
+        assertSize(4, target);
+        assertTrue(target.containsAll(true, false, false, true));
+    }
+
+    @Test
+    default void RichIterable_flatCollectByte()
+    {
+        RichIterable<ByteBytePair> iterable =
+                this.newWith(
+                        PrimitiveTuples.pair((byte) 1, (byte) 2),
+                        PrimitiveTuples.pair((byte) 3, (byte) 4));
+
+        ByteArrayList target = new ByteArrayList();
+        iterable.flatCollectByte((Function<ByteBytePair, ByteIterable>) each -> ByteArrayList.newListWith(
+                each.getOne(),
+                each.getTwo()), target);
+
+        assertSize(4, target);
```

Comment:
Use an equality test with the target bag, which will cover size and containsAll.